### PR TITLE
Remove `build_sysroot` folder

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
With this, the top-level directory is finally clean.

### CODE REVIEW

File: build_system/src/prepare.rs [hunk 1]
```diff
@@ -116,17 +89,11 @@ fn prepare_libcore(
     patches.sort();
     for file_path in patches {
         println!("[GIT] apply `{}`", file_path.display());
-        let path = Path::new("../..").join(file_path);
+        let path = Path::new("../../..").join(file_path);
```
Comment (antoyo) [inline build_system/src/prepare.rs:92]: Would it be clearer to use an absolute path?

File: build_system/src/test.rs [hunk 2]
```diff
@@ -908,28 +830,17 @@ fn should_remove_test(file_path: &Path) -> Result<bool, String> {
...
-            "//@ error-pattern:",
-            "//@ build-fail",
-            "//@ run-fail",
-            "-Cllvm-args",
-            "//~",
-            "thread",
-        ]
-        .iter()
-        .any(|check| line.contains(check))
+        if ["//@ error-pattern:", "//@ build-fail", "//@ run-fail", "-Cllvm-args", "//~", "thread"]
```
Comment (antoyo) [inline build_system/src/test.rs]: I would prefer to keep the original formatting, so please use `#[rustfmt::skip]` or whatever.

**Comment by antoyo** [comment]
The CI failed, possible because you would need to adjust the `--sysroot` argument that we send to rustc.

File: build_system/src/prepare.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply build_system/src/prepare.rs:92]: Hard to say...

File: build_system/src/test.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply build_system/src/test.rs]: Sounds good to me.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -63,7 +63,7 @@ jobs:
         git config --global user.name "User"
         ./y.sh prepare
         # FIXME(antoyo): we cannot enable LTO for stdarch tests currently because of some failing LTO tests using proc-macros.
-        echo -n 'lto = "fat"' >> build_sysroot/Cargo.toml
+        echo -n 'lto = "fat"' >> build/build_sysroot/Cargo.toml
```
Comment (antoyo) [inline .github/workflows/release.yml]: Could the CI failure due to the fact that this change is overwritten by the fact that you now copy this file?

Comment (GuillaumeGomez) [reply .github/workflows/release.yml]: Hum indeed.

File: .github/workflows/stdarch.yml [hunk 4]
```diff
@@ -89,12 +89,12 @@ jobs:
     - name: Run stdarch tests
       if: ${{ !matrix.cargo_runner }}
       run: |
-        cd build_sysroot/sysroot_src/library/stdarch/
-        CHANNEL=release TARGET=x86_64-unknown-linux-gnu CG_RUSTFLAGS="-Ainternal_features" ../../../../y.sh cargo test
+        cd build/build_sysroot/sysroot_src/library/stdarch/
+        CHANNEL=release TARGET=x86_64-unknown-linux-gnu CG_RUSTFLAGS="-Ainternal_features" ../../../../../y.sh cargo test
```
Comment (antoyo) [inline .github/workflows/stdarch.yml]: This is starting to have many `../`. Would it work to use `--manifest-path` instead (without changing directory)?

File: build_system/src/test.rs [hunk 5]
```diff
@@ -157,8 +127,10 @@ impl TestArg {
                     show_usage();
                     return Ok(None);
                 }
-                x if runners.contains_key(x) => {
-                    test_arg.runners.insert(x.into());
+                x if runners.contains_key(x)
+                    && !test_arg.runners.iter().any(|runner| runner == x) =>
```
Comment (antoyo) [inline build_system/src/test.rs:131]:
```suggestion
                    && !test_arg.runners.contains(x) =>
```

File: build_system/src/test.rs [hunk 6]
```diff
@@ -913,23 +838,18 @@ fn should_remove_test(file_path: &Path) -> Result<bool, String> {
             "//@ build-fail",
             "//@ run-fail",
             "-Cllvm-args",
-            "//~",
-            "thread",
+            "//~", "thread",
```
Comment (antoyo) [inline build_system/src/test.rs]:
```suggestion
            "//~",
            "thread",
```

File: .github/workflows/stdarch.yml (hunk 4 above)
Comment (GuillaumeGomez) [reply .github/workflows/stdarch.yml]: It's worth a try.

File: build_system/src/test.rs (hunk 5 above)
Comment (GuillaumeGomez) [reply build_system/src/test.rs:131]: It's what I did first, but:

```
error[E0308]: mismatched types
    --> src/test.rs:131:51
     |
131  |                     && !test_arg.runners.contains(x) =>
     |                                          -------- ^ expected `&String`, found `&str`
     |                                          |
     |                                          arguments to this method are incorrect
     |
     = note: expected reference `&std::string::String`
                found reference `&str`
```

**Comment by antoyo** [comment]
Thanks for your contribution.

**Review by antoyo — APPROVED** [review]

**Pull request merged**